packages/marko-web-html-sitemap/routes/index.js

```diff
@@ -0,0 +1,150 @@
+      },
+    );
+  }));
+
+  app.get(`${ROOT_ALIAS}/:year(\\d{4})/:month(\\d{2})`, asyncRoute(async (req, res) => {
+    const { year, month } = req.params;
+    const startOfMonth = moment(`${year}-${month}-01`, FORMAT);
+    if (!startOfMonth.isValid()) throw invalidDate();
+    const endOfMonth = startOfMonth.clone().endOf('month');
+    const now = moment().endOf('day');
+    const after = startOfMonth.format(FORMAT);
+    if (startOfMonth > now) throw dateNotFound();
+    const before = (endOfMonth > now ? now : endOfMonth).format(FORMAT);
+    const input = {
+      before,
+      after,
+      format: 'days',
+    };
+    const variables = { input };
+    const { data } = await req.apollo.query({ query: ALL_PUBLISHED_CONTENT_DATES, variables });
+    const nodes = data.allPublishedContentDates.map((node) => {
+      const alias = `${ROOT_ALIAS}/${year}/${month.toLocaleString('en-US', { minimumIntegerDigits: 2, useGrouping: false })}`;
+      const publishedContentNode = node;
+      publishedContentNode.alias = `${alias}/${node.day.toLocaleString('en-US', { minimumIntegerDigits: 2, useGrouping: false })}`;
+      return publishedContentNode;
+    });
+    return res.marko(
+      dateListTemplate,
+      {
+        nodes,
+        year,
+        month: startOfMonth.format('MMMM'),
+        displayType: 'day',
+      },
+    );
+  }));
+
+  app.get(`${ROOT_ALIAS}/:year(\\d{4})`, asyncRoute(async (req, res) => {
+    const { year } = req.params;
+    const now = moment().endOf('day');
+    if (year > now.format('year')) throw dateNotFound();
+    const nowFormatted = now.format(FORMAT);
+    const end = `${year}-12-31`;
+    const after = `${year}-01-01`;
+    const before = nowFormatted < end ? nowFormatted : end;
+    const input = {
+      before,
+      after,
+      format: 'months',
+    };
+    const variables = { input };
+    const { data } = await req.apollo.query({ query: ALL_PUBLISHED_CONTENT_DATES, variables });
+    const nodes = data.allPublishedContentDates.map((node) => {
+      const alias = `${ROOT_ALIAS}/${year}`;
+      const publishedContentNode = node;
+      publishedContentNode.alias = `${alias}/${node.month.toLocaleString('en-US', { minimumIntegerDigits: 2, useGrouping: false })}`;
+      return publishedContentNode;
+    });
+    return res.marko(
+      dateListTemplate,
+      {
+        nodes,
+        year,
+        displayType: 'month',
+      },
+    );
+  }));
+
+  app.get(`${ROOT_ALIAS}`, asyncRoute(async (req, res) => {
+    const now = moment().endOf('day');
+    const year = moment().format('YYYY');
+    const nowFormatted = now.format(FORMAT);
+    const end = `${year}-12-31`;
+    const before = nowFormatted < end ? nowFormatted : end;
+    const input = {
+      before,
+      format: 'years',
+    };
+    const variables = { input };
+    const { data } = await req.apollo.query({ query: ALL_PUBLISHED_CONTENT_DATES, variables });
+    const nodes = data.allPublishedContentDates.map((node) => {
```

Comment:
```suggestion
    const nodes = getAsArray(data, "allPublishedContentDates").map((node) => {
```

You never know what sneaky results might come back!